<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1346</width>
    <height>999</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QListWidget" name="listWidget">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>670</y>
      <width>771</width>
      <height>321</height>
     </rect>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>511</width>
      <height>901</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>EA parameters</string>
     </attribute>
     <layout class="QGridLayout" name="gridLayout">
      <item row="2" column="0" colspan="2">
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Adult selection :</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="6" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Group size (K)  :</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QSpinBox" name="tournamentGroupSizeAdult">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>1000000</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Sigma :</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QDoubleSpinBox" name="tournamentSigmaAdult">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Over production :</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QRadioButton" name="globalRadioA">
           <property name="text">
            <string>Global selection strategy</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QComboBox" name="comboBox">
           <item>
            <property name="text">
             <string>Fiteness proportional</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Sigma scaling</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Rank selection</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QRadioButton" name="localStratAdult">
           <property name="text">
            <string>Local selection strategy </string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="overProduction">
           <property name="maximum">
            <number>200</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Mixing :</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QCheckBox" name="mixingCheckBox">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>Rank SP :</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QDoubleSpinBox" name="rankSP">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.001000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="numberOfGeneration">
        <property name="maximum">
         <number>100000000</number>
        </property>
        <property name="value">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Adult ppool size :</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Number of generation :</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="adultPoolSize">
        <property name="maximum">
         <number>100000000</number>
        </property>
        <property name="value">
         <number>200</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Children selection</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="3" column="1">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Group size (k) :</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QRadioButton" name="globalRadioA_2">
           <property name="text">
            <string>Global selection strategy</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QRadioButton" name="localStratChild">
           <property name="text">
            <string>Local selection strategy</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="comboBox_2">
           <item>
            <property name="text">
             <string>Fiteness proportional</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Sigma scaling</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Rank selection</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Sigma :</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QSpinBox" name="tournamentGroupSizeChild">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>100000000</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QDoubleSpinBox" name="tournamentSigmaChild">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
        <zorder>globalRadioA_2</zorder>
        <zorder>localStratChild</zorder>
        <zorder>label_8</zorder>
        <zorder>label_9</zorder>
        <zorder>tournamentGroupSizeChild</zorder>
        <zorder>tournamentSigmaChild</zorder>
        <zorder>comboBox_2</zorder>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Genetic operator</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="1" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Mutation probability :</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Crossover probability :</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="mutationRate">
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="maximum">
            <double>20.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.001000000000000</double>
           </property>
           <property name="value">
            <double>0.003000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="crossOverRate">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.500000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Problem parameters</string>
     </attribute>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="9" column="0">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Alphabet size :</string>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="QSpinBox" name="alphabetSize">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>26</number>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QRadioButton" name="surpriseSeqLoc">
        <property name="text">
         <string>Local suprising sequence</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QRadioButton" name="lolz">
        <property name="text">
         <string>LOLZ problem</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QRadioButton" name="oneMatrix">
        <property name="text">
         <string>One matrix problem </string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QSpinBox" name="bitLength">
        <property name="maximum">
         <number>300</number>
        </property>
        <property name="value">
         <number>40</number>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Bit length :</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="oneMatrixR">
        <property name="text">
         <string>Random matrix problem</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QRadioButton" name="surpriseSeqGlob">
        <property name="text">
         <string>Global surprising sequence</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Lolz parameter</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QSpinBox" name="lolzZ">
        <property name="value">
         <number>21</number>
        </property>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>Length :</string>
        </property>
       </widget>
      </item>
      <item row="10" column="1">
       <widget class="QSpinBox" name="lengthSuprise">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>53</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QPushButton" name="goButton">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>940</y>
      <width>131</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Go !</string>
    </property>
   </widget>
   <widget class="QCustomPlot" name="plot" native="true">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>50</y>
      <width>771</width>
      <height>601</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="abort">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>940</y>
      <width>112</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Abort</string>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>globalRadioA</sender>
   <signal>toggled(bool)</signal>
   <receiver>tournamentGroupSizeAdult</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>250</x>
     <y>275</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>globalRadioA</sender>
   <signal>toggled(bool)</signal>
   <receiver>tournamentSigmaAdult</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>276</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>localStratAdult</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>160</x>
     <y>341</y>
    </hint>
    <hint type="destinationlabel">
     <x>177</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>globalRadioA_2</sender>
   <signal>toggled(bool)</signal>
   <receiver>tournamentGroupSizeChild</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>515</y>
    </hint>
    <hint type="destinationlabel">
     <x>375</x>
     <y>616</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>globalRadioA_2</sender>
   <signal>toggled(bool)</signal>
   <receiver>tournamentSigmaChild</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>220</x>
     <y>523</y>
    </hint>
    <hint type="destinationlabel">
     <x>315</x>
     <y>655</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>localStratChild</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_2</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>587</y>
    </hint>
    <hint type="destinationlabel">
     <x>189</x>
     <y>559</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
